### PR TITLE
fix(API): Time periods exceptions (issue on 00:00-00:00 range)

### PR DESCRIPTION
**Pull request by asalhi-ext**
Fix Time periods exceptions  on range  : 0:00-00:00 

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

<h2> How this pull request can be tested ? </h2>

check [ticket](https://centreon.atlassian.net/browse/MON-149029)

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php [hunk 1]
```diff
@@ -31,6 +31,11 @@
  */
 class TimeRange implements \Stringable
 {
+    /**
+     * @var string TIME_RANGE_FULL_DAY_ALIAS The time range for the entire day in DOC
+     */
+    public const TIME_RANGE_FULL_DAY_ALIAS = '00:00-00:00';
```
Comment (Drago37) [inline centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php:37]: private no ?

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php [hunk 2]
```diff
@@ -42,6 +47,15 @@ class TimeRange implements \Stringable
...
     public function __construct(string $timeRange)
     {
+
+        $resolvedTimeRangeAlias = $this->resolveTimeRangeAlias($timeRange);
+        $timeRange = $resolvedTimeRangeAlias['value'];
+        if ($resolvedTimeRangeAlias['isFullDayAlias']) {
+            $this->timeRange = $timeRange;
+
+            return;
+        }
```
Comment (Drago37) [inline centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php:57]: why do you need to return an array from resolveTimeRangeAlias ? you can change the value of $this->timeRAnge directly in resolveTimeRangeAlias without using an array ;)

**Review by Drago37 — CHANGES_REQUESTED** [review]

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php (hunk 2 above)
Comment (asalhi-ext) [reply centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php]: resolved

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php (hunk 1 above)
Comment (asalhi-ext) [reply centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php]: resolved

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php [hunk 3]
```diff
@@ -42,6 +47,13 @@ class TimeRange implements \Stringable
...
+
+        $resolvedTimeRangeAlias = $this->resolveTimeRangeAlias($timeRange);
+        if ($resolvedTimeRangeAlias) {
+            return;
+        }
+
+        $timeRange = $this->timeRange;
         Assertion::minLength($timeRange, 11, 'TimeRange::timeRange');
 
         if (! $this->isValidTimeRangeFormat($timeRange)) {
```
Comment (Drago37) [inline centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php:59]: Seen together : 
* delete your method resolveTimeRangeAlias()
* move in the method isValidTimeRangeFormat(), the following code :  `Assertion::minLength($timeRange, 11, 'TimeRange::timeRange');` at the begining
* in the method isValidTimeRangeFormat(), before :  `Assertion::minLength($timeRange, 11, 'TimeRange::timeRange');` write this code :
```
if (empty($timeRange)) {
    return true;
}
```
* delete your changes in the unit test
* enjoy ! :D

**Review by Drago37 — CHANGES_REQUESTED** [review]
Seen together : 
* delete

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php (hunk 3 above)
Comment (asalhi-ext) [reply centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php]: resolved

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php (hunk 2 above)
Comment (Drago37) [reply centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php]: ok

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php (hunk 3 above)
Comment (Drago37) [reply centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php]: good

**Review by Drago37 — APPROVED** [review]

File: centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php [hunk 4]
```diff
@@ -79,6 +82,8 @@ public function getRanges(): array
      */
     private function isValidTimeRangeFormat(string $timeRange): bool
     {
+        Assertion::minLength($timeRange, 11, 'TimeRange::timeRange');
```
Comment (callapa) [inline centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php:85]: By moving the assertion, we hide the fact that the isValidTimeRangeFormat method can raise an exception
```suggestion
```

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by callapa — APPROVED** [review]

**Review by Drago37 — APPROVED** [review]

**Pull request merged**